example/example.cpp

```diff
@@ -0,0 +1,61 @@
+#include "osrm/json_container.hpp"
+#include "osrm/engine_config.hpp"
+#include "osrm/route_parameters.hpp"
+#include "osrm/osrm.hpp"
+
+#include <string>
+#include <functional>
+
+int main(int argc, const char *argv[])
+{
+    if (argc < 2)
+    {
+        std::cerr << "Error: Not enough arguments." << std::endl
+                  << "Run " << argv[0] << " data.osrm" << std::endl;
+    }
+    try
```

Comment:
Just flatten this example, as in

    int main() try {

    } catch(const std::exception& e) {
        std::cerr << "Error: " << e.what();
        return EXIT_FAILURE;
    }